--- /opt/perl-5.14.1/lib/site_perl/5.14.1/Data/Dump.pm	2010-10-19 19:20:43.000000000 +0100
+++ /opt/perl-5.14.0/lib/site_perl/5.14.0/Data/Dump.pm	2011-06-09 16:55:51.000000000 +0100
@@ -13,10 +13,12 @@
 $DEBUG = 0;
 
 use overload ();
-use vars qw(%seen %refcnt @dump @fixup %require $TRY_BASE64 @FILTERS $INDENT);
+use vars qw(%seen %refcnt @dump @fixup %require $TRY_BASE64 @FILTERS $INDENT $QUOTE_KEYS $SINGLE_QUOTE);
 
 $TRY_BASE64 = 50 unless defined $TRY_BASE64;
 $INDENT = "  " unless defined $INDENT;
+$QUOTE_KEYS = 0 unless $QUOTE_KEYS;
+$SINGLE_QUOTE = 0 unless $SINGLE_QUOTE;
 
 sub dump
 {
@@ -303,6 +305,7 @@
 	}
 
 	my $quote;
+  $quote++ if $QUOTE_KEYS;
 	for my $key (@orig_keys) {
 	    next if $key =~ /^-?[a-zA-Z_]\w*\z/;
 	    next if $key =~ /^-?[1-9]\d{0,8}\z/;
@@ -533,7 +536,7 @@
   local($_) = $_[0];
   # If there are many '"' we might want to use qq() instead
   s/([\\\"\@\$])/\\$1/g;
-  return qq("$_") unless /[^\040-\176]/;  # fast exit
+  return ( $SINGLE_QUOTE ? qq('$_') : qq("$_") ) unless /[^\040-\176]/;  # fast exit
 
   s/([\a\b\t\n\f\r\e])/$esc{$1}/g;
 
@@ -543,6 +546,7 @@
   s/([\0-\037\177-\377])/sprintf('\\x%02X',ord($1))/eg;
   s/([^\040-\176])/sprintf('\\x{%X}',ord($1))/eg;
 
+  return qq('$_') if $SINGLE_QUOTE;
   return qq("$_");
 }
 
